### PR TITLE
Load json cyclops fix - faster tests - fixes of various test failures

### PR DESCRIPTION
**Pull request by egillax**
Here is the PR I mentioned yesterday for you to review @jreps . 

Features
- Changed cyclops coefficient to use a data.frame with columns ```betas``` and ```covariateIds``` instead of a named vector. 
- Adapted test for external validation to follow real world behavior, almost always you load the model from disk and then validate externally. This would have caught the json saving bug. 
- Added ```quiet=True``` to all AUC calculations, silencing the message that comes everytime you calculate the AUC.

Bugfixes
- Added priorCoefs to documentation since it was failing the R CMD check. 
- Fixed a small bug where the externalValidate wasn't picking up the population settings which caused a test failure 
- There was a hardcoded ```.json``` in the sklearn test, now since that has been turned of it broke a test - I changed it to ```.pkl``` 
- A small bug causing a warning since the KNN metaData wasn't saved correctly.
- Fixed a small bug caused by the cyclops coefficient being a dataframe, the intercept wasn't uploading to the database correctly.
- In ```loadPlpResults``` , it would force the class of the object to be ```runPlp```. Sometimes the loaded object was a ```validatePlp``` object, then the class would be overwritten. Now it only adds the ```runPlp``` class if the class attribute is missing.
- When uploading results to database in validateMultiplePlp , there was an error becauce it was looking for a ```plpResult``` folder when the folder with the results was named ```validateResult```. Now it looks for a folder with ```Result``` in it and picks that.
- In ```externalValidatePlp``` the ```targetId``` wasn't correctly assigned, so it was null - I fixed that.
- The univariateFeatureSelection was failing because the ```y``` object the sklearn function was receiving was wrong.
- When the Cyclops model didn't fit, there was an error when creating the coefficient dataframe because there was only one coefficient ```c(0)``` with a ```NULL``` name. I changed the name in those cases to an empty string.
- There was an occasionally failing test in preprocessing. This happened when there was a redundant covariate to remove but no rare covariates to remove. I think the unique function was then being used incorrectly in ```PreprocessingData.R``` and as a consequence the deleted covariates weren't removed from the covariateRef. 
- Another rare failing test - in the plotting of the preference score. Very rarely I think there was some numerical instability issue which caused the ```preferenceScore``` not to be monotonically increasing/decreasing after sorting by the ```prediction$value```. This caused an error when plotting the ```preferenceScore```

Test improvements
- Test weren't running locally because some tests were for uploading to databases using the github secrets. When running github actions there is an environment variable ```CI``` which is set to ```true```. So I used that to automatically detect if test are running locally or on actions, then it skips those tests that cannot be run locally automatically. 
- Updated the actions workflow to use the new Hades one, got it from DatabaseConnector.  It's a bit faster and simpler. 
- For the KNN test I now sample 400 samples for it to run on to be faster. I do this by adding a population object to the plpData object. I also suppress warnings when the KNN fits 100% on the training set.
- I made the plpData objects created in the test helper smaller. It's now 1000 samples, which in my opinion is enough, then you have about 30 events per fold and 30 in the test set.
- I removed the plpDataBig object, it was only used to test the batchRestrict for Andromeda. IMO this isn't really necessary and the functionality can easily be tested with smaller data objects.
- For testing the permutation feature importance. Instead of using all the covariates now I sample 10 of them to calculate the importance on.  

Now I run the tests locally in around 5 minutes on my linux laptop. The actions now run in about 37 minutes instead of an 1 hour and 15 minutes. I still see occasionally failing tests and warnings but leave that for future work.

### CODE REVIEW

**Comment by jreps** [comment]
Excellent - thanks for all these fixes! :D :D :D
